projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/collection/write-collection-field.component.ts

```diff
@@ -332,6 +335,13 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
     return !!id && !this.getCollectionPermission(this.caseField, 'allowDelete');
   }
 
+  public checkComplexFieldReadOnly(): boolean {
+    return this.caseField?.field_type?.collection_field_type?.complex_fields?.every((complexField) => {
```

Comment:
This will work for a single level, but it may be insufficiently recursive. Does it work if the complex field contains collections, or nested complex types, or collections of complex nested types?